R/router.R

```diff
@@ -119,18 +137,42 @@ create_router_callback <- function(root, routes) {
       }
     )
 
-    observeEvent(session$userData$shiny.router.page(), {
+    shiny::observeEvent(session$userData$shiny.router.page(), {
       page_path <- session$userData$shiny.router.page()$path
       log_msg("shiny.router main output. path: ", page_path)
-      shiny::removeUI("#router-page-content")
-      shiny::insertUI(
-        "#router-page-wrapper", "afterBegin",
-        shiny::div(id = "router-page-content", routes[[page_path]][["ui"]])
-      )
+      session$sendCustomMessage("switch-ui", page_path)
     })
   }
 }
 
+router_server <- function(router) {
+  create_router_callback(router$root, router$routes)
+}
+
+router_ui <- function(router) {
```

Comment:
docs please

Reply:
I'll add this but this is internal function as well.